File: cs50x/mario.c
#include <cs50.h>
#include <stdio.h>

void print_row(int spaces, int bricks);

int main(void)
{
    // prompt user for input
    int height;

    do
    {

        height = get_int("Enter the height of the pyramid(1 - 8)? \n");
    }

    while (height < 1 || height > 8);

    // print a height of the pyramid
    {
        for (int i = 0; i < height; i++)
        {
            int spaces = height - (i + 1); // leading spaces
            int bricks = i + 1;            // bricks in this row
            print_row(spaces, bricks);
        }
    }
}

void print_row(int spaces, int bricks)
{

    // print spaces
    for (int i = 0; i < spaces; i++)
    {
        printf(" ");
    }
    // print bricks
    for (int i = 0; i < bricks; i++)
    {
        printf("#");
    }
    // print gap
    printf("  ");
    // print bricks
    for (int i = 0; i < bricks; i++)
    {
        printf("#");
    }
    printf("\n"); // prints newline
}

File: cs50x/mario.py
def print_row(spaces, bricks):
    # print leading spaces
    print(" " * spaces, end="")

    # print left pyramid bricks
    print("#" * bricks, end="")

    # print gap
    print("  ", end="")  # two spaces

    # print right pyramid bricks
    print("#" * bricks)


# 1. Get valid height
while True:
    height = input("Enter height between 1 and 8: ")

    # Check if input is numeric
    if height.isnumeric():
        height_int = int(height)
        # Check if in valid range
        if 1 <= height_int <= 8:
            break   # valid input, exit loop
    print("Invalid! Enter height between 1 and 8: ")

# 2. Print the pyramid row by row
for i in range(1, height_int + 1):
    spaces = height_int - i        # leading spaces for right alignment
    bricks = i                     # number of bricks
    print_row(spaces, bricks)
